cmd/oras/internal/option/parser.go

```diff
@@ -28,26 +29,29 @@ type FlagParser interface {
 
 // Parse parses applicable fields of the passed-in option pointer and returns
 // error during parsing.
-func Parse(cmd *cobra.Command, optsPtr interface{}) error {
-	return rangeFields(optsPtr, func(fp FlagParser) error {
-		return fp.Parse(cmd)
-	})
+func Parse(cmd *cobra.Command, optsPtr any) error {
+	for parser := range fields[FlagParser](optsPtr) {
+		if err := parser.Parse(cmd); err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
-// rangeFields goes through all fields of ptr, optionally run fn if a field is
-// public AND typed T.
-func rangeFields[T any](ptr any, fn func(T) error) error {
-	v := reflect.ValueOf(ptr).Elem()
-	for i := 0; i < v.NumField(); i++ {
-		f := v.Field(i)
-		if f.CanSet() {
-			iface := f.Addr().Interface()
-			if opts, ok := iface.(T); ok {
-				if err := fn(opts); err != nil {
-					return err
+// fields returns an iterator that yields all fields of the given struct that
+// implement the given interface.
+func fields[T any](ptr any) iter.Seq[T] {
+	return func(yield func(T) bool) {
+		v := reflect.ValueOf(ptr).Elem()
+		for i := 0; i < v.NumField(); i++ {
```

Comment:
nit: Looks like this can be simplified to:
```suggestion
		for i := range v.NumField()
```

Reply:
Good one as this expr was introduced in go 1.22.